Moves add_tester_to_groups and remove_tester_from_groups to Spaceship::TestFlight::Group

taquitos opened the pull request:
So that it can be reused in fastlane/boarding project

### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
Adding testers to app groups is shared with boarding, so instead of duplicating logic, I moved it into Spaceship's Application class

taquitos commented:
This probably needs some changes since referencing Spaceship::TestFlight:: in Application is bad form.

snatchev commented:
It might be better to move this code into either `Spaceship::TestFlight::Tester` or `Spaceship::TestFlight::Group`

Commit:
Rebased on most recent changes
Moved add/remove users to/from group logic to class methods in Spaceship::TestFlight::Group

taquitos commented:
@snatchev I moved the logic into Spaceship::TestFlight::Group, that seems like the best place for it.

fastlane-bot-helper commented:
<!--
  0 Errors
  1 Warning: Pull Request is Work in Progre...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Pull Request is Work in Progress</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

Commit:
Updated class method vs instance method

taquitos commented on spaceship/lib/spaceship/test_flight/group.rb:
```diff
@@ -48,26 +48,26 @@ def self.internal_group(app_id: nil)
     # is not enough to add the tester to a group. If this isn't done the next request would fail.
     # This is a bug we reported to the iTunes Connect team, as it also happens on the iTunes Connect UI on 18. April 2017
-    def self.add_tester!(tester)
+    def add_tester!(tester)
```
I accidentally changed this

taquitos commented on spaceship/lib/spaceship/test_flight/group.rb:
```diff
@@ -48,26 +48,26 @@ def self.internal_group(app_id: nil)
     end
 
-    def self.remove_tester!(tester)
+    def remove_tester!(tester)
```
I accidentally changed this

ohayon reviewed (approved):
💯
